daemon/src/model/cfd.rs

```diff
@@ -449,7 +449,7 @@ impl Cfd {
     #[allow(dead_code)]
     pub const CET_TIMELOCK: u32 = 12;
 
-    pub fn handle(&self, event: CfdStateChangeEvent) -> Result<CfdState> {
+    pub fn handle(&mut self, event: CfdStateChangeEvent) -> Result<CfdState> {
```

Comment:
I think it might even be easier to understand to consume `self` and return an entirely new `Cfd`?

Reply:
I agree, but this is a quite intrusive change and I would do it in a separate refactoring. I'll create a ticket.

There are several considerations we *might* want to on top of that: 

- Not save "state by id" because it is error prone. Instead always change a cfd and then save that cfd.
- Move **all** state transitions onto the cfd's `handle` function. At the moment there are still several early stage states that are just transitioned in the cfd actor without the cfd `handle` function being involved.

Reply:
https://github.com/comit-network/hermes/issues/200